### DISCUSSION TITLE
Examples for using AdvertisementFilter in BleakScannerDotNet

### DISCUSSION THREAD

**Discussion by marvincabuenas** [hbldh/bleak#590] [Q&A] [answered]
Hello,

I would just like to ask if I can get some code examples on how to use the AdvertisementFilter for BleakScannerDotNet. I am currently involved in a project that aims to detect the BLE device name without knowing the Bluetooth Address. If someone can point me to some code, it will be very much appreciated.

Sincerely,

Marvin Neil

**Comment by dlech** [comment, marked as answer]
I would suggest using [BleakScanner.find_device_by_filter()](https://bleak.readthedocs.io/en/latest/api.html#bleak.backends.scanner.BaseBleakScanner.find_device_by_filter). Here is an [example program](https://github.com/hbldh/bleak/blob/55a2d34cc96bb842be278485794806704caa2d2c/examples/scanner_byname.py). (However if you need to match two things and one is in the advertising data packet and one is in the scan response packet, it won't work on Windows currently - on my todo list.)

**Comment by marvincabuenas** [comment]
Hello @dlech 

Thank you for your reply. I'll try first with BleakScanner.find_device_by_filter().

Sincerely,

Marvin Neil
